### DISCUSSION TITLE
Maximum lambda size across separate APIs

### DISCUSSION THREAD

**Discussion by sambecker** [vercel/vercel#4041] [General] [answered]
My APIs, those located in `pages/apis/`, seem to be behaving like they have some shared size limit, not unlike how `pages` and `api`s [used to behave](https://github.com/zeit/now/discussions/4013).

Minimal reproducible repo:
https://github.com/sambecker/now-function-size

One, but not both, of these APIs can be successfully deployed to Zeit Now:
- `test-api.ts` or
- `test-api-alt.ts`

In other words, if I have a `30mb` dependency in one endpoint, and a `25mb` dependency in another endpoint, then both endpoints cannot be deployed (because together they have a size of `50mb+`), however each can be deployed individually.

It is my understanding that each API endpoint represents a discrete lambda with its own dependency bundle/size limitations. If this is incorrect, please let me know!

cc @paulogdm @styfle

**Comment by sambecker** [comment]
Deployment error logs in case they're helpful:

```
9:51:12.236  Compiled successfully.
09:51:12.236  Automatically optimizing pages...
09:51:12.843  Page                                                           Size     First Load JS
09:51:12.843  ┌ ○ /                                                          6.94 kB        64.9 kB
09:51:12.843  ├ ○ /404                                                       3.15 kB        61.1 kB
09:51:12.843  ├ λ /api/test-api
09:51:12.843  ├ λ /api/test-api-alt
09:51:12.843  └ λ /test-page                                                 293 B          58.2 kB
09:51:12.843  + First Load JS shared by all                                  57.9 kB
09:51:12.843    ├ static/pages/_app.js                                       957 B
09:51:12.843    ├ chunks/e2e2e76c8bf507b2d0f54f2e5110e7a0e38cd0f7.92878c.js  10.3 kB
09:51:12.843    ├ chunks/framework.0f140d.js                                 40 kB
09:51:12.843    ├ runtime/main.7e74ef.js                                     5.95 kB
09:51:12.843    └ runtime/webpack.b65cab.js                                  746 B
09:51:12.844  λ  (Lambda)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
09:51:12.844  ○  (Static)  automatically rendered as static HTML (uses no initial props)
09:51:12.844  ●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
09:51:12.895  Done in 7.76s.
09:51:18.943  Traced Next.js serverless functions for external files in: 6031.273ms
09:51:32.783  Compressed shared serverless function files: 13839.791ms
09:51:32.947  All serverless functions created in: 162.795ms
09:51:32.988  Error: The Serverless Function "api/test-api-alt" is 51.63mb which exceeds the maximum size limit of 50mb. Learn more: https://zeit.co/docs/v2/platform/limits#serverless-function-size
09:51:34.613  Done with "package.json"
```

**Comment by mcsdevv** [comment, marked as answer]
Hey @sambecker, this behavior is expected currently. However, this will change when this PR is merged: https://github.com/zeit/now/pull/4102

**Reply by sambecker** [reply]
So exciting! Looking forward to that being deployed. Would make my current project viable.

**Reply by sambecker** [reply]
@ChrisLusted can't confirm this was fixed but it looks like it should have been according to @mcsdevv ...

**Reply by ChrisLusted** [reply]
@sambecker thanks for the message. It doesn’t appear to be. I managed to work around it by moving the api folder out of the next pages folder and into the root. Not ideal but its working for the moment

**Reply by gregavola** [reply]
@ChrisLusted How did you move them outside of pages, and route them correctly? This seems to me the issue for me, as all my function are the capacity for some reason.

**Comment by styxlab** [comment]
Running into the same issue. @ChrisLusted: What are the (unwanted) side effects of moving the `api` folder out of the `pages` folder?
